main.py

```diff
@@ -394,97 +440,217 @@ def export_assets(request):
                 output_config=output_config,
             )
 
-            client.export_assets(request=request)
-            logging.info(f"Asset export triggered for content type: {content_type}")
+            # Capture the returned operation
+            operation = client.export_assets(request=request)
+
+            # Extract bucket_name and path from OUTPUT_BUCKET
+            gcs_prefix = "gs://"
+            if OUTPUT_BUCKET.startswith(gcs_prefix):
+                bucket_name = OUTPUT_BUCKET[len(gcs_prefix) :].split("/")[0]
+                path = f"asset_export_v2_{timestamp}/{content_type}"
+            else:
+                logging.error(f"Invalid GCS URI: {OUTPUT_BUCKET}")
+                raise ValueError(f"Invalid GCS URI: {OUTPUT_BUCKET}")
+
+            full_gcs_path = f"{bucket_name}/{path}/operation_name.txt"
+
+            # Write the operation name to GCS
+            bucket = storage_client.bucket(bucket_name)
+            blob = bucket.blob(f"{path}/operation_name.txt")
+            blob.upload_from_string(operation.operation.name)
+
+            logging.info(
+                f"Asset export triggered for content type: {content_type}. Operation name: {operation.operation.name} saved to GCS."
+            )
+
+            create_cloud_task(full_gcs_path)
 
         except Exception as e:
-            logging.error(f"Failed to export content type {content_type}. Error: {e}")
-            logging.error(traceback.format_exc())
+            logging.critical(
+                f"Failed to export content type {content_type}. Error: {e}",
+                exc_info=True,
+            )
             return f"Failed to export content type {content_type}. Error: {e}", 500
 
     return "Asset export triggered", 200
 
 
-# Triggered by a change in a storage bucket
-@functions_framework.cloud_event
-def gcs_to_pubsub(cloud_event: CloudEvent):
-    """
-    Function triggered by a change in a storage bucket. When a change is detected,
-    it fetches the blob from the bucket, and if it passes checks, sends its contents
-    to a Pub/Sub topic.
+def create_cloud_task(blob_path):
+    # Initialize client
+    client = tasks_v2.CloudTasksClient()
+    project = PARENT.split("/")[1]
+    queue_path = client.queue_path(project, GCP_REGION, TASK_QUEUE)
+
+    # Construct the URL for the cloud function. This URL will be hit by Cloud Tasks.
+    url = f"https://{GCP_REGION}-{project}.cloudfunctions.net/{GCS_TO_PUBSUB_CLOUD_FUNCTION}"
```

Comment:
Let's have terraform send the entire cloud function URI which removed the need to have GCP_REGION as an environment variable